packages/client/hmi-client/src/views/TA2Playground.vue

```diff
@@ -695,74 +607,58 @@ export default defineComponent({
 				const modelData = await resp.data;
 				modelId = modelData.id;
 				console.log(`Model ID: ${modelId}`); // currently required for testing
-			}
 
-			// Reset current nodes and edges
-			g.nodes = [];
-			g.edges = [];
+				const juliaNodes: { name: string; type: string }[] = [];
+				const juliaEdges: { source: string; target: string }[] = [];
 
-			const nodeHeight = 20;
-			const nodeWidth = 20;
-			let nodeX = 0;
-			let nodeY = 0;
-			// Nodes
-			for (let i = 0; i < model.S.length; i++) {
-				const aNode = model.S[i];
-				nodeX += 30;
-				nodeY += 30;
-				this.addNode(
-					`s-${i + 1}`,
-					aNode.sname.toString(),
-					nodeX,
-					nodeY,
-					nodeHeight,
-					nodeWidth,
-					NodeType.Species,
-					createFlag
-				);
-			}
-			// Move Transitions 100 to the right of S
-			nodeX = 100;
-			nodeY = 0;
-			for (let i = 0; i < model.T.length; i++) {
-				const aTransition = model.T[i];
-				nodeX += 30;
-				nodeY += 30;
-				this.addNode(
-					`t-${i + 1}`,
-					aTransition.tname.toString(),
-					nodeX,
-					nodeY,
-					nodeHeight,
-					nodeWidth,
-					NodeType.Transition,
-					createFlag
-				);
-			} // end T
-
-			// Edges
-			for (let i = 0; i < model.I.length; i++) {
-				const iEdges = model.I[i];
-				const sourceID = `s-${iEdges.is}`;
-				const transitionID = `t-${iEdges.it}`;
-				this.addEdgeID(sourceID, transitionID, createFlag);
-			}
-			for (let i = 0; i < model.O.length; i++) {
-				const oEdges = model.O[i];
-				const sourceID = `s-${oEdges.os}`;
-				const transitionID = `t-${oEdges.ot}`;
-				this.addEdgeID(transitionID, sourceID, createFlag);
+				// add Nodes
+				for (let i = 0; i < model.S.length; i++) {
```

Comment:
All those for loop could be `Array.map()`

Reply:
Done - hope more readable now!